open_instruct/model_utils.py

```diff
@@ -404,14 +404,26 @@ def save_with_accelerate(
     output_dir: str,
     use_lora: bool = False,
     model_attribute_to_save: Optional[str] = None,
+    chat_template_name: str = "tulu",
 ) -> None:
     """`model_attribute_to_save` is for used to save PPO's policy instead of the full model"""
     # set the generation config to an empty setting to be safe.
     # we usually do greedy decoding for generation, so this should be okay.
     # otherwise, we get an error thrown at save time.
-    model.generation_config = transformers.GenerationConfig(
-        temperature=None, top_p=None, eos_token_id=tokenizer.eos_token_id, bos_token_id=tokenizer.bos_token_id
-    )
+    if "olmo" in chat_template_name:
+        # New chat template has no bos token, and two eos tokens: <|im_end|> and <|endoftext|>
+        model.generation_config = transformers.GenerationConfig(
```

Comment:
does this saving change also need to be added to the grpo/ppo scripts?

Reply:
I think it does, only for from-zero RL experiments. if we're RL'ing on top of a previously SFT'd/DPO'd model, it should already have the updated generation config, but I updated all the save functions for PPO/GRPO to be safe.